generator/src/googleapis/codegen/template_objects.py

```diff
@@ -128,6 +128,57 @@ def __init__(self, def_dict, api, parent=None, wire_name=None,
       self.SetTemplateValue('description',
                             self.ValidateAndSanitizeComment(self.StripHTML(d)))
 
+  def ComputeNonDuplicatedName(self, original_class_name):
+    from googleapis.codegen.api import Resource
+    from googleapis.codegen.api import Method
+
+    class_name = original_class_name
+    parent_path = self.parentPath()
+
+    def getNonDuplicatedClassName(class_name):
+      if str(class_name).startswith('Child'):
+        return 'Grand' + class_name
+      elif str(class_name).startswith('GrandChild') or str(class_name).startswith('Great'):
+        return 'Great' + class_name
+      else:
+        return 'Child' + class_name
+
+    # Prepend the first ancestor's name with an underscore if there is any
```

Comment:
I think this comment is not valid anymore now that we are using the `[Great][Grand][Child]` pattern?

Reply:
Fixed